### ISSUE TITLE
Linux compilation issue

### ISSUE THREAD

**Issue by Sanaki** [nwhitehead/pzretro#4]
Decided to give this core a spin just to test, ran into an issue with compilation on Mint 19.3 (Ubuntu 18.04 LTS essentially). Error occurred with both GCC and clang, with multiple versions.
```
./src/js.o: In function `js::Context::start_thread(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
js.cpp:(.text+0xd33): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
Makefile:771: recipe for target 'puzzlescript_libretro.so' failed
```
I resolved this issue locally (in a suboptimal way) via
`export CC="gcc -pthread" CXX="g++ -pthread"`
after a quick search led me here:
https://stackoverflow.com/a/1665110

I'm sure there's a cleaner way to resolve this on the code side, but I'm hoping seeing my solution will help you identify what the correct approach would be.

**Comment by nwhitehead** [comment]
It's weird that this behavior of GCC changed over time. Not sure why the flag is no longer needed.

I added the flag to the new simplified Makefile build. Hopefully that fixes the issue.

**Comment by Sanaki** [comment]
I'm still seeing the error from a basic bare `make` without the changed CC/CXX.

**Comment by nwhitehead** [comment]
Added the flag to the linking part in latest commit, that fixed it in my Linux Mint 19.3 docker container test at least.

**Comment by Sanaki** [comment]
That resolved it. Thanks!
